src/circuits/mimc/mimc_hash.tcc

```diff
@@ -0,0 +1,94 @@
+// DISCLAIMER:
+// Content taken and adapted from:
+// https://github.com/HarryR/ethsnarks/blob/master/src/gadgets/mimc.hpp
+
+#ifndef __ZETH_MIMC_HASH_TCC__
+#define __ZETH_MIMC_HASH_TCC__
+
+namespace libzeth {
+
+template<typename FieldT>
+MiMC_hash_gadget<FieldT>::MiMC_hash_gadget(
+		libsnark::protoboard<FieldT> &in_pb,
+		const libsnark::pb_variable<FieldT> in_iv,
+		const std::vector<libsnark::pb_variable<FieldT>>& in_messages,
+    const libsnark::pb_variable<FieldT> in_out,
+		const std::string &in_annotation_prefix
+	) :
+		libsnark::gadget<FieldT>(in_pb, in_annotation_prefix),
+		messages(in_messages),
+		iv(in_iv),
```

Comment:
why prefix the inputs by `in_` ? `iv(iv)` should work well

Reply:
Modified